src/amo/components/FeaturedAddons/index.js

```diff
@@ -38,19 +59,39 @@ export class FeaturedAddonsBase extends React.Component {
     }
   }
 
+  loadDataIfNeeded(nextProps = {}) {
+    const { addonType: lastAddonType } = this.props;
+    const {
+      errorHandler, dispatch, loading, results,
+    } = { ...this.props, ...nextProps };
+
+    const nextParams = nextProps && nextProps.params ?
```

Comment:
Sure, I added some comments. This component is suffering from tracking URL state in two places which is why it's a bit confusing. It could be simplified once we track URL state purely in redux: https://github.com/mozilla/addons-frontend/issues/181